app/src/test/java/org/pesmypetcare/mypetcare/controllers/exercise/TestTrAddWalk.java

```diff
@@ -0,0 +1,121 @@
+package org.pesmypetcare.mypetcare.controllers.exercise;
+
+import com.google.android.gms.maps.model.LatLng;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.pesmypetcare.httptools.utilities.DateTime;
+import org.pesmypetcare.mypetcare.features.pets.Pet;
+import org.pesmypetcare.mypetcare.features.pets.PetRepeatException;
+import org.pesmypetcare.mypetcare.features.pets.events.InvalidPeriodException;
+import org.pesmypetcare.mypetcare.features.pets.events.exercise.walk.Walk;
+import org.pesmypetcare.mypetcare.features.users.NotPetOwnerException;
+import org.pesmypetcare.mypetcare.features.users.User;
+import org.pesmypetcare.mypetcare.services.StubPetManagerService;
+import org.pesmypetcare.usermanager.datacontainers.pet.GenderType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Albert Pinto
+ */
+public class TestTrAddWalk {
+    private static final int COORDINATES_NUMBER = 5;
+    private User user;
+    private List<Pet> pets;
+    private List<LatLng> coordinates;
+    private TrAddWalk trAddWalk;
+
+    @Before
+    public void setUp() throws PetRepeatException {
+        user = new User("johnDoe", "johndoe@gmail.com", "1234");
+        pets = new ArrayList<>();
+        coordinates = new ArrayList<>();
+
+        Pet pet = getDinkyPet();
+        user.addPet(pet);
+        pets.add(pet);
+
+        for (int actual = 0; actual < COORDINATES_NUMBER; ++actual) {
+            coordinates.add(new LatLng(actual, actual));
+        }
+
+        trAddWalk = new TrAddWalk(new StubPetManagerService());
+    }
+
+    @Test(expected = NotPetOwnerException.class)
+    public void shouldNotAddWalkingToNonOwnerPet() throws NotPetOwnerException, InvalidPeriodException,
+        ExecutionException, InterruptedException {
+        trAddWalk.setUser(new User("johnSmith", "johnsmith@gmail.com", "5678"));
+        trAddWalk.setPets(pets);
+        trAddWalk.setName("A walk for the neighbourhood");
+        trAddWalk.setDescription("Walking on the near surroundings of our home");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "Walking on the near surroundings of our home" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5561369)